tasks/scripting/src/main/java/org/infinispan/scripting/impl/ScriptingTaskEngine.java

```diff
@@ -39,6 +40,12 @@ public String getName() {
       return tasks;
    }
 
+   @Override
+   public CompletionStage<List<Task>> getTasksAsync() {
+      BlockingManager blockingManager = scriptingManager.cacheManager.getGlobalComponentRegistry().getComponent(BlockingManager.class);
+      return blockingManager.supplyBlocking(this::getTasks, "ScriptingTaskEngine - getTasksAsync");
```

Comment:
isn't `Cache.stream()` non blocking? could it be used instead of sending to a new thread?

Reply:
The call to `stream` is, but any subsequent method invocations are not in this case `forEach` is not. That is what the ClusterPublisherManager is for, but we don't have a nice API for it yet, so I figured this was simpler for now.